src/components/AddressDetailExplorer.js

```diff
@@ -0,0 +1,528 @@
+    loadingTokens: true,
+    errorMessage: '',
+    warningRefreshPage: false,
+    warnMissingTokens: 0,
+    selectedTokenMetadata: null,
+    metadataLoaded: false,
+    addressTokens: {},
+    txCache: {},
+    showReloadDataButton: false,
+    showReloadTokenButton: false,
+  }
+
+  componentDidMount() {
+    // Expects address on URL
+    this.updateAddress(this.props.match.params.address);
+
+    WebSocketHandler.on('network', this.handleWebsocket);
+  }
+
+  componentWillUnmount() {
+    WebSocketHandler.removeListener('network', this.handleWebsocket);
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.match.params.address !== this.props.match.params.address) {
+      // Address on the URL changed
+      this.pagination.clearOptionalQueryParams();
+      this.updateAddress(this.props.match.params.address);
+      return;
+    }
+
+    const queryParams = this.pagination.obtainQueryParams();
+
+    // Do we have new URL params?
+    if (!isEqual(this.state.queryParams, queryParams)) {
+      if (queryParams.token !== this.state.queryParams.token && queryParams.token !== null) {
+        // User selected a new token, so we must go to the first page (clear queryParams)
+        this.pagination.clearOptionalQueryParams();
+        // Need to get newQueryParams because the optional ones were cleared
+        // Update state to set the new selected token on it
+        // If we don't update this state here we might execute a duplicate request
+        const newQueryParams = this.pagination.obtainQueryParams();
+        this.setState({ queryParams: newQueryParams }, () => {
+          this.reloadTokenData(newQueryParams.token);
+        });
+        return;
+      }
+
+      // update the query params state then fetch the new page
+      this.setState({ queryParams, loadingHistory: true }, () => {
+        // Fetch new data, unless query params were cleared and we were already in the most recent page
+        this.getHistoryData(+queryParams.page);
+      });
+    }
+  }
+
+  /**
+   * Called when 'network' ws message arrives
+   * If it's a new tx message update data, in case is necessary
+   *
+   * @param {Object} wsData Data from websocket
+   */
+  handleWebsocket = (wsData) => {
+    if (wsData.type === 'network:new_tx_accepted') {
+      if (this.shouldUpdate(wsData, false) && !this.state.warningRefreshPage) {
+        // If the search address is in one of the inputs or outputs
+        this.setState({ warningRefreshPage: true });
+      }
+    }
+  }
+
+  /**
+   * Check if address is valid and then update the state and get data from full node
+   * If not valid show error message
+   *
+   * @param {Object} address New searched address to update state
+   */
+  updateAddress = (address) => {
+    this.setState({ queryParams: this.pagination.obtainQueryParams() }, () => {
+      if (hathorLib.transaction.isAddressValid(address)) {
+        this.setState({
+          address,
+          loadingTokens: true,
```

Comment:
While this is true, I believe we should set them on both places.
This would prevent some UI "errors" like showing the table empty and incomplete summary then showing the loading icons.
And the `reloadData` and `reloadTokenData` also need to set the loading states because there are other places which use them (e.g. selected token changed) and having this state management in the method prevents duplicating this code everywhere.